server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java

```diff
@@ -155,4 +155,9 @@ public void createField(DocumentParserContext context, BytesRef tsid) {
         BytesRef uidEncoded = Uid.encodeId(context.id());
         context.doc().add(new Field(NAME, uidEncoded, FIELD_TYPE));
     }
+
+    @Override
+    public String documentDescription(SourceToParse source) {
```

Comment:
Maybe add a comment here saying that we don't have an ID yet because it's generated from the content of the document?